### PR TITLE
Adds tests for DisplayImage (#8)

### PR DESCRIPTION
**Pull request by entrylevelcs**
Added a test for the DisplayImage and updated the README.md to reflect the addition of the tests so that it can still build the executable without the test files affecting it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -72,7 +72,7 @@ To build the project use the command:
   * Assemble the main classes  
     * `./gradlew classes`    
   * Assemble and test the project
-    * `./gradlew build`   
+    * `./gradlew build -x test`
```
Comment (jody) [inline README.md:75]: Why is it necessary to exclude `test` from the build command in order to "Assemble and test the project"?

**Review by jody — CHANGES_REQUESTED** [review]
Please remove the `imagelab.jar` update from this PR.

File: README.md (hunk 1 above)
Comment (entrylevelcs) [reply README.md:75]: We were getting an error where the filters list was not showing up unless we excluded the test from that command, but after running through it again it seems to work so the "-x test" is actually not needed and we will remove it.

**Comment by entrylevelcs** [comment]
Removed the updated imagelab.jar and the addition to the README.md as both were not needed.

File: imagelab/DisplayImageTest.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    public void constructorWithNullImageShouldHaveNullValue() {
+    	DisplayImage imageOne = new DisplayImage(null, "Title");
+    }
+
+    /**
+     * The title of the DisplayImage should not be equal to an incorrect title.
+     */
+    @Test
+    public void titleShouldNotEqualWrongTitle() {
+    	//ImageLab labOne = new ImageLab();
```
Comment (jody) [inline imagelab/DisplayImageTest.java:40]: Please remove this commented-out code remnant.

File: imagelab/DisplayImageTest.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    	DisplayImage imageOne = new DisplayImage(providerOne, "Title");
+    	assertNotEquals(imageOne.getTitle(), "Ditsdle");
+    }
+
+    /**
+     * The title of the DisplayImage should be equal to the correct title.
+     */
+    @Test
+    public void titleShouldBeEqualToCorrectTitle() {
+    	//ImageLab labOne = new ImageLab();
```
Comment (jody) [inline imagelab/DisplayImageTest.java:51]: Please remove this commented-out code remnant.

File: imagelab/DisplayImageTest.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+import static org.junit.Assert.*;
+
+public class DisplayImageTest {
+    /**
+     * Sets up the test fixture.
+     * (Called before every test case method.)
+     */
+    @Before
+    public void setUp() {
+       ImageLab labOne = new ImageLab();
```
Comment (jody) [inline imagelab/DisplayImageTest.java:16]: Please remove this. (This appears to create and initialize an unused local variable in a `setUp` method.)

**Review by jody — CHANGES_REQUESTED** [review]

**Review by jody — APPROVED** [review]

**Comment by jody** [comment]
Ready to be merged, thank you!!
Just need to determine if this also closes User Story #8.
The acceptance criteria for User Story #8 requires 85% code coverage.
Is there documentation of the amount of code coverage provided by these tests?

**Comment by entrylevelcs** [comment]
Using the jacocotestcoverage that is provided, the coverage is listed as 93% but the .gitignore ignores the build directory that contains the report showing the 93% coverage. Should we pull the report out of the build directory so the other contributors can see this report?

**Comment by jody** [comment]
> Using the jacocotestcoverage that is provided, the coverage is listed as 93% ... Should we pull the report out of the build directory so the other contributors can see this report?

That's fine!
You could just put a screen shot of the test coverage results into the discussion.
I'll do that for you here in this comment.

Screen Shot 2020-12-04 at 8 12 48 PM

While running that report I also noticed that there are 4 convention violations reported by Checkstyle.  I've included that report below.  (The report shows each identified shortcoming twice.)  Please fix the code to address those shortcomings.

Screen Shot 2020-12-05 at 3 19 20 PM

**Review by jody — CHANGES_REQUESTED** [review]
We just need to address the Checkstyle coding convention reports, then this should be OK to merge.

**Comment by entrylevelcs** [comment]
Fixed the DisplayImageTest.java to be consistent with coding convention report.
no errors

**Review by jody — APPROVED** [review]
All good!

**Comment by jody** [comment]
Thanks for the responsiveness and quality!

**Pull request merged**